blockchain/stake/staketx.go

```diff
@@ -589,8 +590,8 @@ func CheckSStx(tx *wire.MsgTx) error {
 	}
 
 	// Ensure that the first output is tagged OP_SSTX.
-	if txscript.GetScriptClass(tx.TxOut[0].Version, tx.TxOut[0].PkScript) !=
-		txscript.StakeSubmissionTy {
+	if !standalone.IsStakeOpCodeScript(tx.TxOut[0].Version,
+		[]txscript.ScriptClass{txscript.StakeSubmissionTy}, tx.TxOut[0].PkScript) {
```

Comment:
This seems quite wasteful in forcing a slice to be created just to satisfy the API.  Also, script classes really only apply for purposes of standardness, so I don't think the API should take that at all.  Instead, I would much prefer separate explicit funcs exported.  For example:

* `func IsTicketPurchaseScript(version uint16, script []byte) bool`
* `func IsVoteScript(version uint16, script []byte) bool`
* `func IsRevocationScript(version uint16, script []byte) bool`

Internally to the `standalone` package, you can have a common unexported function that is used to avoid code duplication.

Not only will it make the code easier to reason about for consumers, it will be more efficient.

I won't comment on all the other cases, but they'd all be updated to call the correct funcs accordingly.